ui/components/ImageLink.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from "react";
+import styled from "styled-components";
+import Link from "./Link";
+import Text from "./Text";
+
+export const convertImage = (image: string) => {
+  const split = image.split("/");
+  const prefix = split.shift();
+  let url = "";
+
+  const makeUrl = (url: string, parts: string[]) => {
+    parts.forEach((part, index) => {
+      if (index === split.length - 1) url += part;
+      else url += `${part}/`;
+    });
+    return url;
+  };
```

Comment:
It might be worth it moving this function to `lib/utils.ts` and adding tests for it (there are similar sample tests for the `isHTTP` function in `utils.test.ts`). It seems like a rather standalone utility function.

And besides adding tests would help to see how the resulting URL returned from the function is different from the input URL.

If it is moved to that file, instead of using local constants within function body directly, the values of these constants could be passed to the function as arguments.